src/nodes/Collection.ts

```diff
@@ -207,4 +214,90 @@ export abstract class Collection extends NodeBase {
         )
     }
   }
+
+  /** Merge another collection into this */
+  merge(source: Collection|Document): void {
+    let sourceNode: Node;
+
+    if (source instanceof Document) {
+      if (isNode(source.contents)) {
+        sourceNode = source.contents
+      } else {
+        throw new Error(`Cannot merge Document content type "${typeof source.contents}" into Document`)
+      }
```

Comment:
> Rather than source instanceof Document, use isDocument(source)
Oh, I missed that one. Fixed it.

> there shouldn't be a need to duplicate the isNode() check in the two branches.
Correct. I made this more elegant now.